### PR TITLE
Add changes from rector rule BinaryOpNullableToInstanceofRector

### PR DESCRIPTION
**Pull request by Harfusha**
https://github.com/cakephp/cakephp/issues/17756

### CODE REVIEW

File: src/Console/CommandFactory.php [hunk 1]
```diff
@@ -44,7 +44,7 @@ public function __construct(?ContainerInterface $container = null)
      */
     public function create(string $className): CommandInterface
     {
-        if ($this->container && $this->container->has($className)) {
+        if ($this->container instanceof \Cake\Core\ContainerInterface && $this->container->has($className)) {
```
Comment (ADmad) [inline src/Console/CommandFactory.php:47]: Are instanceof checks more efficient than implicit boolean conversion?

Comment (othercorey) [reply src/Console/CommandFactory.php:47]: > Are instanceof checks more efficient than implicit boolean conversion?

This change suggests it's not a container interface which doesn't seem right.

Comment (ADmad) [reply src/Console/CommandFactory.php:47]: > This change suggests it's not a container interface which doesn't seem right.

Yes, it's misleading. I would rather keep this unchanged.

**Comment by Harfusha** [comment]
Adding this rule to ignore list

**Pull request closed without merging**